### PR TITLE
Update remote-apis-sdks

### PR DESCRIPTION
**Pull request by peterebden**
This unforks remote-apis so we're fully in line with their protocol on compression. Updates remote-apis-sdks (still with a bunch of custom changes because they don't support this yet, although I've sent most of it back upstream now).

Have tested locally by rebuilding mettle against the local instance, which shook out a few small bugs.

I've bumped a major version (beta) because this won't roll out in one go; we need to roll out Elan and Flair, then the Mettle workers to maximise intelligibility of the Capabilities messages.

### CODE REVIEW

File: rexclient/client.go [hunk 1]
```diff
@@ -25,35 +24,24 @@ const CompressionThreshold = 1024
...
+		&client.TreeSymlinkOpts{Preserved: true},
+		client.CompressedBytestreamThreshold(CompressionThreshold),
+		client.UploadCompressionClassifier(client.DetectArchiveUploads),
+		client.UsePackName(PackName),
+	)
 	if err != nil {
 		log.Error("Error initialising remote execution client: %s", err)
 		return nil, err
 	}
-
```
Comment (Garbett1) [inline rexclient/client.go:38]: So this all goes away with the new compression arguments I guess?

File: elan/rpc/rpc.go [hunk 2]
```diff
@@ -571,42 +575,52 @@ func (s *server) readBlob(ctx context.Context, key string, offset, length int64)
...
-	buf := make([]byte, 0, digest.SizeBytes)
-	if compressed {
-		return s.compressor.EncodeAll(b, buf), nil
-	}
-	b, err = s.decompressor.DecodeAll(b, buf)
-	if err != nil {
-		return nil, err
-	}
-	blobsServed.WithLabelValues(batchLabel(batched, false), compressorLabel(compressed), compressorLabel(!compressed)).Inc()
+	blobsServed.WithLabelValues(batchLabel(batched, false), compressorLabel(compressed), compressorLabel(compressed)).Inc()
```
Comment (Garbett1) [inline elan/rpc/rpc.go:618]: `compressorLabel(compressed),compressorLabel(!compressed)`

File: elan/rpc/eclient.go [hunk 3]
```diff
@@ -56,11 +56,13 @@ func (e *elanClient) UpdateActionResult(req *pb.UpdateActionResultRequest) (*pb.
 	return e.s.UpdateActionResult(ctx, req)
 }
 
-func (e *elanClient) UploadIfMissing(entries []*uploadinfo.Entry) error {
+func (e *elanClient) UploadIfMissing(entries []*uploadinfo.Entry, compressors []pb.Compressor_Value) error {
```
Comment (Garbett1) [inline elan/rpc/eclient.go:59]: Does changing this signature mean that all the Please clients will fail for a short period of time if we roll Mettle first?

Comment (peterebden) [reply elan/rpc/eclient.go:59]: Not this particular case, no. This interface is only relevant within the process.

The upload/download blob stuff should all be compatible here, the main bit that's not is the `Capabilities` stuff (which I think we can work around by ordering the rollout; we could probably avoid it by doing it in several more steps but I don't think it's worth it)

**Review by Garbett1 — APPROVED** [review]
Good to go once the metric label is updated.

File: elan/rpc/rpc.go (hunk 2 above)
Comment (peterebden) [reply elan/rpc/rpc.go]: Hmmmm on some thought I've set the second one to not applicable. They're "compressor used" and "compressor preferred", in this new world there is no preferred compressor when serving batched blobs.

**Pull request merged**